server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java

```diff
@@ -83,7 +83,14 @@ public void removeListener(TransportConnectionListener listener) {
 
     @Override
     public void openConnection(DiscoveryNode node, ConnectionProfile profile, ActionListener<Transport.Connection> listener) {
-        delegate.openConnection(node, profile, listener);
+        delegate.openConnection(
```

Comment:
Separate PR open for this here, which will need to merge first: https://github.com/elastic/elasticsearch/pull/92584

Reply:
Does this PR depend on this change?

Reply:
Yup this PR depends on https://github.com/elastic/elasticsearch/pull/92584 (since there are scenarios where a connection is actually remote that we don't currently detect -- the linked PR addresses this) -- I will wait for approval on 92584, merge that, then merge this current PR.